sdks/java/core/src/test/java/org/apache/beam/sdk/testing/PAssertTest.java

```diff
@@ -117,6 +124,42 @@ public void registerByteSizeObserver(
     }
   }
 
+  @Test
+  public void testSuccessOrFailureEncodedDecoded() throws IOException {
+    AssertionError error = null;
+    try {
+      assertEquals(0, 1);
+    } catch (AssertionError e) {
+      error = e;
+    }
+    SuccessOrFailure success = PAssert.SuccessOrFailure.success();
+    SuccessOrFailure failure = PAssert.SuccessOrFailure.failure(
+        new PAssertionSite(error.getMessage(), error.getStackTrace()));
+
+    ByteArrayOutputStream sharedOutStream = new ByteArrayOutputStream();
+    AvroCoder<SuccessOrFailure> coder = AvroCoder.of(SuccessOrFailure.class);
+    coder.encode(success, sharedOutStream, null);
+    InputStream sharedInStream = new ByteArrayInputStream(sharedOutStream.toByteArray());
+    SuccessOrFailure res = coder.decode(sharedInStream, null);
```

Comment:
Look at CoderUtils, I believe, to make this easier to test.

@kennknowles can elaborate on coders.